power_grid_model_c/power_grid_model/include/power_grid_model/sparse_ordening.hpp

```diff
@@ -0,0 +1,243 @@
+// SPDX-FileCopyrightText: 2022 Contributors to the Power Grid Model project <dynamic.grid.calculation@alliander.com>
+//
+// SPDX-License-Identifier: MPL-2.0
+
+#pragma once
+#ifndef POWER_GRIdx_MODEL_MATH_SOLVER_SPARSE_ORDENING_HPP
+#define POWER_GRIdx_MODEL_MATH_SOLVER_SPARSE_ORDENING_HPP
+
+#include "power_grid_model.hpp"
+
+#include <algorithm> // remove and remove_if
+#include <map>
+#include <string>
+#include <utility>
+#include <vector>
+
+namespace power_grid_model {
+
+namespace detail {
+inline void remove_element_vector_pair(Idx& u, std::vector<std::pair<Idx, Idx>>& dgd) {
+    Idx i = 0;
+    while (i < dgd.size()) {
+        if (dgd[i].first == u) {
+            dgd.erase(dgd.begin() + i);
+            break;
+        } else {
+            i++;
+        }
+    }
```

Comment:
improve implementation, e.g.

```suggestion
    std::erase_if(dgd, [&u](auto const& v) { v.first == u});
```